src/viz/Ensight_Translator.i.hh

```diff
@@ -434,58 +457,92 @@ void Ensight_Translator::write_geom(const uint32_t part_num,
 
   size_t ndim = pt_coor.ncols(0);
   size_t nvertices = vertices.size();
+  size_t g_nvertices = nvertices;
+  if (d_decomposed)
+    rtt_c4::global_sum(g_nvertices);
 
   // output part number and names
-  d_geom_out << "part" << endl;
-  d_geom_out << part_num << endl;
-  d_geom_out << part_name << endl;
-  d_geom_out << "coordinates" << endl;
-  d_geom_out << int(nvertices) << endl; // #vertices in this part
+  if (rtt_c4::node() == 0) {
+    d_geom_out << "part" << endl;
+    d_geom_out << part_num << endl;
+    d_geom_out << part_name << endl;
+    d_geom_out << "coordinates" << endl;
+    d_geom_out << int(g_nvertices) << endl; // #vertices in this part
+  }
+  d_geom_out.flush();
 
   // output the global vertex indices and form ens_vertex.  Enight demands that
   // vertices be numbered from 1 to the number of vertices *for this part*
   // (nvertices).  Argghhh.  ens_vertex maps our local vertex index to a vertex
   // in [1,nvertices].
 
+  // Form global cell and vertex indices.  These are the same as their local
+  // index, in this case.
+  sf_int node_verts(rtt_c4::nodes(), 0);
+  node_verts[rtt_c4::node()] = static_cast<int>(nvertices);
+
+  // set parallel offset for cell and vertices numbers
+  if (d_decomposed)
+    rtt_c4::global_sum<int>(&node_verts[0], rtt_c4::nodes());
+  int local_vert_offset = 0;
+  for (auto n = 0; n < rtt_c4::nodes(); n++) {
+    if (n < rtt_c4::node()) {
+      local_vert_offset += node_verts[n];
+    }
+  }
```

Comment:
Could this logic to get the vert_offsets maybe be replaced with a prefix_sum() from c4? I used these to calculate rank-wise global cell offsets in the parmetis PR if you want to look at an example. They're declared in src/c4/C4_Functions.hh :)

Reply:
Cool. I will try this. I didn't know such a function existed.

Reply:
This worked great.